src/Glpi/OAuth/Server.php

```diff
@@ -177,43 +182,84 @@ public static function getScopeDescriptions(): array
 
     public static function generateKeys(): void
     {
-        $private_key_path = GLPI_CONFIG_DIR . '/oauth.pem';
-        $public_key_path = GLPI_CONFIG_DIR . '/oauth.pub';
-        if (!file_exists($private_key_path) && !file_exists($public_key_path)) {
-            $config = [
-                'digest_alg'       => 'sha512',
-                'private_key_bits' => 2048,
-                'private_key_type' => OPENSSL_KEYTYPE_RSA,
-            ];
-            $success = false;
-            $error = null;
-            $res = openssl_pkey_new($config);
-            if ($res && openssl_pkey_export_to_file($res, $private_key_path)) {
-                // Export public key to the public key file
-                $pubkey = openssl_pkey_get_details($res);
-                if ($pubkey !== false && file_put_contents($public_key_path, $pubkey['key']) === strlen($pubkey['key'])) {
-                    if (chmod($private_key_path, 0o660) && chmod($public_key_path, 0o660)) {
-                        $success = true;
-                    } else {
-                        $error = 'Unable to set permissions on the generated keys';
-                    }
-                } else {
-                    $error = 'Unable to export public key';
-                }
-            } else {
-                $error = 'Unable to generate keys';
-            }
-
-            if (!$success) {
-                // Key files didn't exist before and an error occured. We should try removing any that were created to be able to retry later
-                if (file_exists($private_key_path)) {
-                    unlink($private_key_path);
-                }
-                if (file_exists($public_key_path)) {
-                    unlink($public_key_path);
-                }
-                throw new \RuntimeException($error);
-            }
+        if (
+            file_exists(self::PRIVATE_KEY_PATH)
+            && file_exists(self::PUBLIC_KEY_PATH)
+        ) {
+            // Keys are already generated
+            return;
+        }
+
+        try {
+            // Reset keys in case of partial data (i.e. only the private or only
+            // the public key exist). Should not be able to happen but let's be
+            // safe.
+            self::deleteKeys();
```

Comment:
I'm not sure we should delete in that case. If one file has been dropped by error, this would silently replace it, and therefore cause key related issues.
Throw an exception and let user fix on his side would be more safe I guess.